search/src/main/java/artic/edu/search/SearchResultsSuggestedViewModel.kt

```diff
@@ -2,23 +2,104 @@ package artic.edu.search
 
 import com.fuzz.rx.bindTo
 import com.fuzz.rx.disposedBy
+import edu.artic.db.daos.ArticObjectDao
+import edu.artic.db.daos.ArticSearchObjectDao
 import edu.artic.viewmodel.BaseViewModel
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.Observables
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.Subject
 import javax.inject.Inject
 
-class SearchResultsSuggestedViewModel @Inject constructor(manager: SearchResultsManager) : BaseViewModel() {
-
-    val cells: Subject<List<SearchResultBaseCellViewModel>> = BehaviorSubject.create()
+class SearchResultsSuggestedViewModel @Inject constructor(private val manager: SearchResultsManager,
+                                                          private val searchSuggestionsDao: ArticSearchObjectDao,
+                                                          private val objectDao: ArticObjectDao)
+    : SearchResultsBaseViewModel() {
 
+    private val dynamicCells: Subject<List<SearchResultBaseCellViewModel>> = BehaviorSubject.create()
+    private val suggestedArtworks: Subject<List<SearchResultCircularCellViewModel>> = BehaviorSubject.create()
 
     init {
-        manager.currentSearchResults
-                .map {
-                    it.suggestions.map { SearchResultTextCellViewModel(it) }
-                }.bindTo(cells)
+        setupOnMapSuggestionsBind()
+
+        setupResultsBind()
+
+        Observables
+                .combineLatest(
+                        dynamicCells,
+                        Observable.just(listOf(
+                                SearchResultAmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_restaurant),
+                                SearchResultAmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_lounge),
+                                SearchResultAmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_shop),
+                                SearchResultAmenitiesCellViewModel(R.drawable.ic_icon_amenity_map_restroom),
+                                SearchResultAmenitiesCellViewModel(0))
```

Comment:
I couldn't understand why we need the `SearchResultAmenitiesCellViewModel` without the icon here? Is this for breaking the line for amenities icon? 

I did exactly the same work in `DefaultSearchSuggestionsFragment` for displaying on the map suggestions. Since the UI for **on the map** section for default search suggestions and search suggestions is exactly the same,I think we should share the viewModels.

Reply:
We def should, I thought I only needed to copy small parts of it. but it looks like it was much larger. Hope to do that at a later point during a refactor